hoot-services/src/main/java/hoot/services/controllers/osm/user/UserResource.java

```diff
@@ -340,6 +344,154 @@ public Response getPrivileges(@Context HttpServletRequest request) {
         return Response.ok(json).build();
     }
 
+    /**
+    *
+    * Saves the favorite adv opts for the specified users list
+    *
+    * POST hoot-services/osm/api/0.6/user/saveFavoriteOpts
+    *
+    * @param request
+    * @param favoriteOpts list of objects containing the users id and favorite
+    *  looks like:
+    *      [
+    *        {
+    *        	favorites:
+    *        		{
+    *                   description:  "If true, not only will....",
+    *                   value: true,
+    *                   input: opt.input,
+    *                   type: "boolean",
+    *                   id: AddReviewTagsToFeatures,
+    *                   label: Add Review Tags
+    *        		}
+    *        },
+    *
+    *      ]
+    * @return success status if everything is updated.
+    */
+    @POST
+    @Path("/saveFavoriteOpts")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response saveFavoriteOpts(@Context HttpServletRequest request, String favoriteOpts) {
+
+        try {
+            Users user = Users.fromRequest(request);
+            Long userId = user.getId();
+
+            JSONParser parser = new JSONParser();
+            JSONObject json = (JSONObject) parser.parse(favoriteOpts);
+
+            String getName = (String) json.get("name");
+
+            Object favoritesColumn = createQuery()
+                    .select(users.favoriteOpts)
+                    .from(users)
+                    .where(users.id.eq(userId))
+                    .fetchOne();
+
+            Map<String, String> tags = PostgresUtils.postgresObjToHStore(favoritesColumn);
+
+            String getFavorites = json.get("members").toString();
+
+            if ( !tags.containsKey(getName) ) {
+                tags.put(getName, getFavorites);
+                createQuery().update(users)
+                    .where(users.id.eq(userId))
+                    .set(users.favoriteOpts, tags)
+                    .execute();
+            }
+            else {
+                tags.remove(getName, getFavorites);
```

Comment:
the put() method will overwrite the old value with the new for the specified key so the remove isnt necessary.
Without the remove the if and else can be merged into just a put and update query